### PR TITLE
Cut 0.33.0-rc.0

### PR DESCRIPTION
**Pull request by MichaHoffmann**

### CODE REVIEW

File: docs/release-process.md
Comment (metalmatze) [inline docs/release-process.md:1]: I'd say to make these changes directly to `main` but they'll eventually merge back, so it should be fine.

**Review by metalmatze — APPROVED** [review]
Looks good. 
If this is what needs to be done. I guess there was a bot that writes all the changes into the changelog, right?

**Review by fpetkovski — APPROVED** [review]

File: docs/release-process.md
Comment (yeya24) [reply docs/release-process.md:1]: +1 to have this change in main. Better to be a separate pr

**Review by yeya24 — APPROVED** [review]

File: docs/release-process.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Release shepherd responsibilities:
 
 | Release | Time of first RC     | Shepherd (GitHub handle)      |
 |---------|----------------------|-------------------------------|
-| v0.33.0 | (planned) 2023.10.03 | TBD                           |
```
Comment (squat) [inline docs/release-process.md:26]: this change is causing the docs check to fail

**Review by squat — CHANGES_REQUESTED** [review]
lgtm - the change to the VERSION doc, which is causing CI to fail

**Review by saswatamcode — APPROVED** [review]
LGTM! Just need to fix the docs check. 
I think it might be a bit more convenient to prep `main` first for v0.33 via a PR like: https://github.com/thanos-io/thanos/pull/6617

Otherwise, you might get a bunch of merge conflicts

**Review by saswatamcode — APPROVED** [review]
Thanks! Let's get this rolling.

**Pull request merged**